src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int johnSalary = 0;
+        int andrewSalary = 0;
+        int kateSalary = 0;
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+
+        for (String record : data) {
+            String[] parts = record.split(" ");
+            String dateStr = parts[0];
+            String name = parts[1];
+            int hours = Integer.parseInt(parts[2]);
+            int rate = Integer.parseInt(parts[3]);
+            LocalDate date = LocalDate.parse(dateStr, formatter);
+
+            if (date.isBefore(startDate) || date.isAfter(endDate)) {
+                continue;
+            }
+
+            boolean isEmployeeInList = false;
+            for (String employeeName : names) {
+                if (employeeName.equals(name)) {
+                    isEmployeeInList = true;
+                    break;
+                }
+            }
+            if (!isEmployeeInList) {
+                continue;
+            }
+
+            int salary = hours * rate;
+
+            if (name.equals("John")) {
+                johnSalary += salary;
+            } else if (name.equals("Andrew")) {
+                andrewSalary += salary;
+            } else if (name.equals("Kate")) {
+                kateSalary += salary;
+            }
```

Comment:
The if-else block for calculating salaries is hardcoded for specific names (John, Andrew, Kate). This is not scalable or flexible. Instead, consider using a `Map<String, Integer>` to store the salaries and update the salary for each employee dynamically based on the `names` array.